src/Compilers/CSharp/Portable/SourceGeneration/IncrementalGeneratorInitializationContextExtensions.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.SourceGeneration;
+
+internal static partial class IncrementalGeneratorInitializationContextExtensions
+{
+    private static readonly char[] s_nestedTypeNameSeparators = new char[] { '+' };
+
+    /// <summary>
+    /// Returns all syntax nodes of type <typeparamref name="T"/> if that node has an attribute on it that could
+    /// possibly bind to the provided <paramref name="fullyQualifiedMetadataName"/>. <paramref
```

Comment:
> ... that possibly bind ...

This language seems inconsistent with the semantic model check at the end of this method.

Reply:
true :)